src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -240,15 +236,14 @@ private static void ConvertMultiLineToSingleLineSpans(SourceText text, Segmented
             bool supportsVisualStudioExtensions,
             IReadOnlyDictionary<string, int> tokenTypesToIndex)
         {
-            using var _ = ArrayBuilder<int>.GetInstance(classifiedSpans.Count, out var data);
-
             // We keep track of the last line number and last start character since tokens are
             // reported relative to each other.
             var lastLineNumber = 0;
             var lastStartCharacter = 0;
 
             var tokenTypeMap = SemanticTokensSchema.GetSchema(supportsVisualStudioExtensions).TokenTypeMap;
 
+            var data = new FixedSizeArrayBuilder<int>(5 * classifiedSpans.Count);
```

Comment:
note: teh original capacity passed in was incorrect.  we need 5x the elements when doign semantic tokens.